### PR TITLE
Initial design and development

### PR DESCRIPTION
**Pull request by bkmoovio**

### CODE REVIEW

File: checkDetailAddendumB.go [hunk 1]
```diff
@@ -151,14 +155,15 @@ func (cdAddendumB *CheckDetailAddendumB) MicrofilmArchiveSequenceNumberField() s
...
-	return cdAddendumB.alphaField(cdAddendumB.ImageReferenceKeyLength, 4)
+// LengthImageReferenceKeyField gets the LengthImageReferenceKey field
+func (cdAddendumB *CheckDetailAddendumB) LengthImageReferenceKeyField() string {
+	return cdAddendumB.numericField(cdAddendumB.LengthImageReferenceKey, 4)
 }
 
 // ImageReferenceKeyField gets the ImageReferenceKey field
 func (cdAddendumB *CheckDetailAddendumB) ImageReferenceKeyField() string {
-	return cdAddendumB.alphaField(cdAddendumB.ImageReferenceKey, 34)
+	// ToDo: Check if +1
```
Comment (adamdecaf) [inline checkDetailAddendumB.go]: Did you need to check this still?

File: file.go [hunk 2]
```diff
@@ -75,11 +72,11 @@ func (e *FileError) Error() string {
 type File struct {
 	// ID is a client defined string used as a reference to this record
 	ID string `json:"id"`
-	// FileHeader is an ICL FileHeader
+	// FileHeader is a ileHeader
```
Comment (adamdecaf) [inline file.go]: "a FileHeader" ?

File: validators.go [hunk 3]
```diff
@@ -686,14 +686,63 @@ func (v *validator) isOverrideIndicator(code string) error {
...
+	switch code {
+	case
+		// Preliminary notification
+		1,
+		// Final notification
+		2:
+		return nil
+	}
+	msg := fmt.Sprintf(msgInvalid, "ReturnNotificationIndicator")
+	return errors.New(msg)
```
Comment (adamdecaf) [inline validators.go]: I usually use `fmt.Errorf(msg, args...)` for building error messages.

Edit: Should we include the value on these errors?

File: returnBundle.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+func NewReturnBundle(bh *BundleHeader) *ReturnBundle {
+	rb := new(ReturnBundle)
+	rb.SetControl(NewBundleControl())
+	rb.SetHeader(bh)
+	return rb
+}
+
+// Validate performs X9  validations and format rule checks and returns an error if not Validated
+func (rb *ReturnBundle) Validate() error {
+	return nil
```
Comment (adamdecaf) [inline returnBundle.go]: This was the only func I saw left todo!

File: checkDetailAddendumB.go (hunk 1 above)
Comment (bkmoovio) [reply checkDetailAddendumB.go]: Yes, When I create the test.

File: file.go (hunk 2 above)
Comment (bkmoovio) [reply file.go]: Fixed and will be pushed to initial design and development

File: validators.go (hunk 3 above)
Comment (bkmoovio) [reply validators.go]: Added a ToDo in validators for that.

File: returnBundle.go (hunk 4 above)
Comment (bkmoovio) [reply returnBundle.go]: That's correct, 

I also have to do Validate for Bundle, CashLetter, and File.  There will also be additional validations other than correct value, format and inclusion.  For example (# of AddendaRecords equaling ReturnDetail.AddendumCount).   Coming soon..to X9.

**Pull request merged**